.docker/ci/Dockerfile

```diff
@@ -1,50 +1,82 @@
-# moveit/moveit:melodic-ci
-# Sets up a base image to use for running Continuous Integration on Travis
-
-FROM ros:melodic-ros-base
-MAINTAINER Dave Coleman dave@picknik.ai
-
-ENV TERM xterm
-
-# Setup catkin workspace
-ENV CATKIN_WS=/root/ws_moveit
-WORKDIR $CATKIN_WS
-
-# Commands are combined in single RUN statement with "apt/lists" folder removal to reduce image size
-# https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers
-RUN \
-    mkdir src && \
-    cd src && \
-    #
-    # Download moveit source, so that we can get necessary dependencies
-    wstool init --shallow . https://raw.githubusercontent.com/ros-planning/moveit/${ROS_DISTRO}-devel/moveit.rosinstall && \
-    #
-    # Update apt package list as previous containers clear the cache
-    apt-get -qq update && \
-    apt-get -qq dist-upgrade && \
-    #
-    # Install some base dependencies
-    apt-get -qq install -y \
-        # Some source builds require a package.xml be downloaded via wget from an external location
-        wget \
-        # Required for rosdep command
-        sudo \
-        # Preferred build tools
-        python-catkin-tools \
-        clang clang-format-3.9 clang-tidy clang-tools \
-        ccache && \
-    #
-    # Download all dependencies of MoveIt!
-    rosdep update && \
-    rosdep install -y --from-paths . --ignore-src --rosdistro ${ROS_DISTRO} --as-root=apt:false && \
-    #
-    # Remove the source code from this container
-    # TODO: in the future we may want to keep this here for further optimization of later containers
-    cd .. && \
-    rm -rf src/ && \
-    #
-    # Clear apt-cache to reduce image size
-    rm -rf /var/lib/apt/lists/*
-
-# Continous Integration Setting
-ENV IN_DOCKER 1
+# This dockerfile expects proxies to be set via --build-arg if needed
+# It also expects to be contained in the /moveit2 root folder for file copy
+# Example build command:
+# export http_proxy=http://my.proxy.com:80
+# export CMAKE_BUILD_TYPE=Debug
+# docker build -t nav2:latest --build-arg http_proxy --build-arg CMAKE_BUILD_TYPE ./
```

Comment:
moveit2:latest

Reply:
I omitted the org so user building don't blow away what they may have pulled.

Reply:
Makes sense. maybe change nav2->moveit2?

Also I really like that the existing dockerfiles have their label in the documentation (eg https://github.com/ros-planning/moveit2/blob/master/.docker/ci/Dockerfile#L1) do you think you can add a simmilar line up top? It could be something like moveit/moveit2:crystal-circle-ci

Reply:
Rather than have some manual mapping from github branches to dockerhub tags, like `crystal-circle-ci`, I'd recommend we adopt a more programmatic style where release branches and tags on ros-planning/moveit2 github are mapped directly to tag names on ros-planning/moveit2 for CI consumption. There can be procedurally created, while the origin/relation of docker tags to code would be clear and more amenable for CI configs; alluded to in https://github.com/ros-planning/moveit2/pull/44#discussion_r267082024